src/storage/patch/ConvertingPatcher.ts

```diff
@@ -0,0 +1,74 @@
+import { BasicRepresentation } from '../../ldp/representation/BasicRepresentation';
+import type { Representation } from '../../ldp/representation/Representation';
+import { getLoggerFor } from '../../logging/LogUtil';
+import type { RepresentationConverter } from '../conversion/RepresentationConverter';
+import type { RepresentationPatcherInput } from './RepresentationPatcher';
+import { RepresentationPatcher } from './RepresentationPatcher';
+
+/**
+ * A `ConvertingPatcher` converts a document to its `intermediateType`,
+ * sends the result to the wrapped patcher, and then converts back to its original type.
+ * No changes will take place if no `intermediateType` is provided.
+ *
+ * In case there is no resource yet and a new one needs to be created,
+ * the result of the wrapped patcher will be converted to the provided `defaultType`.
+ * In case no `defaultType` is provided, the patcher output will be returned directly.
+ */
+export class ConvertingPatcher extends RepresentationPatcher {
+  protected readonly logger = getLoggerFor(this);
+
+  private readonly patcher: RepresentationPatcher;
+  private readonly converter: RepresentationConverter;
+  private readonly intermediateType?: string;
+  private readonly defaultType?: string;
+
+  /**
+   * @param patcher - Patcher that will be called with the Representation.
+   * @param converter - Converter that will be used to generate intermediate Representation.
+   * @param intermediateType - Content-type of the intermediate Representation if conversion is needed.
```

Comment:
Alternatively: a separate `inConverter` and `outConverter` like we do in `RepresentationConvertingStore`? Again, no strong opinion.

Reply:
True, but creating a custom converter for this would drastically increase the configuration complexity. Even in the `RepresentationConvertingStore` we still add the types (it also takes an `inType` parameter, and "outType" is based on the incoming preferences).